allzpark/control.py

```diff
@@ -1181,25 +1200,50 @@ def _try_resolve_context(req, pkg_name, mode):
 
             self._state["rezApps"][app_request] = rez_pkg
 
+        self._state["rezContexts"] = contexts
+
         self.debug("Resolved all contexts in %.2f seconds" % t.duration)
 
-        # Hide hidden
-        visible_apps = []
+        visible_apps = dict()
+
+        # * Opt-out hidden application
+        # * Find application versions
+        # * Context resolved packages (latest only)
+        all_vers = self._state.retrieve("showAllVersions", False)
         show_hidden = self._state.retrieve("showHiddenApps")
-        for request, package in self._state["rezApps"].items():
-            data = allzparkconfig.metadata_from_package(package)
+        for request, app_pkg in self._state["rezApps"].items():
+            data = allzparkconfig.metadata_from_package(app_pkg)
             hidden = data.get("hidden", False)
 
             if hidden and not show_hidden:
                 continue
 
-            visible_apps += [package]
+            app_versions = [str(v.version) for v in app_ranges[app_pkg.name]]
+            resolved_packages = {
+                pkg: [str(p.version)
+                      for p in (self.find(pkg.name) if all_vers else [pkg])]
+                for pkg in contexts[request].resolved_packages or []
+                if pkg.name not in [app_pkg.name, profile_variant.name]
+            }
+            # profile version should not be changed in Packages view
+            resolved_packages[profile_variant] = [profile_variant]
```

Comment:
All package versions are getting picked here if "Show All Versions" is enabled. No background thread stuff, thought to keep it simple.

Reply:
So long as it keeps the UI responsive. No locking up at any time.

Reply:
Yeah, if there are really massive versions to collect, will only increase profile load time, once being loaded, no extra loading between application selection change.